spec/view_models/journey_navigation_spec.rb

```diff
@@ -0,0 +1,81 @@
+RSpec.describe JourneyNavigation do
+  subject(:journey_navigation) { JourneyNavigation.new(tree, current_step) }
+
+  let(:current_step) { 'other' }
+
+  let(:foo_guide) { instance_double(Guide, content_type: :govspeak, id: double, slug: 'foo', label: 'Foo') }
+  let(:bar_guide) { instance_double(Guide, content_type: :govspeak, id: double, slug: 'bar', label: 'Bar') }
+  let(:baz_guide) { instance_double(Guide, content_type: :govspeak, id: double, slug: 'baz', label: 'Baz') }
+
+  let(:tree) do
+    Tree::TreeNode.from_hash(
+      ['foo', foo_guide] => { ['bar', bar_guide] => { ['baz', baz_guide] => nil } }
+    )
+  end
+
+  describe '.overview' do
+    subject { journey_navigation.overview }
+
+    it { is_expected.to be_a(JourneyNavigation::Step) }
+
+    it 'is not a current step' do
+      expect(subject.current).to be_falsey
+    end
+
+    it 'has the label of the guide at the root of the tree' do
+      expect(subject.label).to eq('Foo')
+    end
+
+    it 'has the URL of the guide at the root of the tree' do
+      expect(subject.url).to eq('/foo')
+    end
+
+    context 'when the overview is the current guide' do
+      let(:current_step) { 'foo' }
+
+      it 'is a current step' do
+        expect(subject.current).to be_truthy
+      end
+    end
+
+    context 'when a child step is the current guide' do
+      let(:current_step) { 'baz' }
+
+      it 'is a current step' do
+        expect(subject.current).to be_truthy
+      end
+    end
+  end
+
+  describe '#steps' do
+    subject { journey_navigation.steps }
+
+    it 'returns steps for each guide' do
+      expect(subject.first).to be_a(JourneyNavigation::Step)
```

Comment:
we should check all steps if we're calling out "each guide"? also perhaps be explicit that its each "child" guide?